### DISCUSSION TITLE
Running large models(Message-IX) on NEOS for free

### DISCUSSION THREAD

**Discussion by PelinaP** [iiasa/message_ix#663] [Q&A]
Can you run an open-source GAMS implementation of the MESSAGE energy system
model on the free Neos server for longer optimization problems? Because of the combination of the python code in jupyter and gams compilation document, is NEOS server a suitable software?

**Comment by behnam-zakeri** [comment]
@PelinaP thanks for asking this question. With a quick check, it seems NEOS server has some requirements for the GAMS files submitted there, including, having the model specified in one single file and avoiding the use of `$INCLUDE`. MESSAGEix has multiple files for compiling the model, which is read one after the other using methods such as `$INCLUDE`. As such, the answer to your question is, no, you cannot use the NEOS server for solving a MESSAGEix model as far as they only accept one file. The workaround will be that you would combine all model files into one, but this requires a good command of GAMS knowledge. I hope this is helpful.
